pkg/webui/console/views/app/index.js

```diff
@@ -122,6 +125,13 @@ const Layout = () => {
   }, [isDrawerOpen, openDrawer, closeDrawer])
   // End of mobile side menu drawer functionality
 
+  useEffect(() => {
```

Comment:
Your earlier iteration of setting the interval within the logic was good. Why did you change it?

Reply:
There two parts to this:
1. The problem changed a bit from the first iteration. Before, we needed to only show a toast when there were new notifications. However, now we also need to update the state in order to display the new notifications dot in the header dropdown. I solved it by adding a `dispatch(notifications.getUnseenNotificationsPeriodicallySuccess())` within the `setInterval`.
2. The problem with the above solution is that apparently, an action is not dispatched within a `setInterval()` in the middleware logic. So although the request happens every 5 minutes, the state is not updated, which means that the notification dot never shows on the header dropdown and the total unseen notifications state doesn't get updated unless the user opens the notifications view. I tried to understand why and used ChatGPT a lot for that, but I couldn't get to the root of it in the end. Whenever I would dispatch an action within a setInterval(), it would never happen. So I looked for other ways to do update the state every 5 minutes. And this was a solution that would update the state. @kschiffer

Reply:
This has likely to do with the fact that you used `createRequestLogic()`, which is our wrapper for redux-logic's `createLogic()` which decorates it with many store and connection specifics, but it also automatically calls `done()` which will cause any subsequent dispatches to be ignored, so this cannot be used for any periodic logics.

So to make this work, please use a custom logic with `createLogic()`, which does not call `done()`. It should then also use the `timeout: 0` option.